packages/core/src/report-components/report-component-data.ts

```diff
@@ -0,0 +1,11 @@
+import { TaskMetaData, ReportComponentType } from '../types';
+
+export abstract class ReportComponentData {
+  meta: TaskMetaData;
+  componentType: ReportComponentType;
+
+  constructor(meta: TaskMetaData, componentType: ReportComponentType) {
+    this.meta = meta;
+    this.componentType = componentType;
+  }
+}
```

Comment:
You can shorten this a bit. If you use the `public` keyword, it will automatically create the public properties and assign them.

```suggestion
export abstract class ReportComponentData {
  constructor(public meta: TaskMetaData, public componentType: ReportComponentType) {
  }
}
```